src/components/carousel/carousel.ts

```diff
@@ -0,0 +1,769 @@
+  public slides!: Array<IgcCarouselSlideComponent>;
+
+  /**
+   * The total number of slides.
+   */
+  public get total(): number {
+    return this.slides.length;
+  }
+
+  /**
+   * The index of the current active slide.
+   */
+  public get current(): number {
+    const index = this.slides.indexOf(this._activeSlide);
+    return index === -1 ? 0 : index;
+  }
+
+  /**
+   * Whether the carousel is in playing state.
+   */
+  public get isPlaying(): boolean {
+    return this._playing;
+  }
+
+  /**
+   * Whether the carousel in in paused state.
+   */
+  public get isPaused(): boolean {
+    return this._paused;
+  }
+
+  @watch('animationType')
+  protected animationTypeChange() {
+    this._context.setValue(this, true);
+  }
+
+  @watch('interval')
+  protected intervalChange() {
+    if (!this.isPlaying) {
+      this._playing = true;
+    }
+
+    this.restartInterval();
+  }
+
+  constructor() {
+    super();
+    this._internals = this.attachInternals();
+
+    this._internals.role = 'region';
+    this._internals.ariaRoleDescription = 'carousel';
+
+    this.addEventListener('pointerdown', this.handlePointerDown);
+    this.addEventListener('pointerenter', this.handlePointerEnter);
+    this.addEventListener('pointerleave', this.handlePointerLeave);
+
+    addKeybindings(this, {
+      ref: this._indicatorsContainerRef,
+      bindingDefaults: { preventDefault: true },
+    })
+      .set(arrowLeft, async () => await this.handleArrowLeft())
+      .set(arrowRight, async () => await this.handleArrowRight())
+      .set(homeKey, async () => await this.handleHomeKey())
+      .set(endKey, async () => await this.handleEndKey());
+
+    addKeybindings(this, {
+      ref: this._prevButtonRef,
+      bindingDefaults: { preventDefault: true },
+    }).setActivateHandler(
+      async () => await this.handleNavigationInteraction('prev')
+    );
+
+    addKeybindings(this, {
+      ref: this._nextButtonRef,
+      bindingDefaults: { preventDefault: true },
+    }).setActivateHandler(
+      async () => await this.handleNavigationInteraction('next')
+    );
+
+    createMutationController(this, {
+      callback: this._observerCallback,
+      filter: [IgcCarouselSlideComponent.tagName],
+      config: {
+        attributeFilter: ['active'],
+        childList: true,
+        subtree: true,
+      },
+    });
+  }
+
+  private handleSlotChange(): void {
+    const index = this.slides.findIndex((slide) => slide.active);
+
+    if (this.total) {
+      index === -1
+        ? this.activateSlide(this.slides[0])
+        : this.activateSlide(this.slides[index]);
+    }
+  }
+
+  private handleIndicatorSlotChange(): void {
+    if (this.hasProjectedIndicators) {
+      this.slides.forEach((slide, i) => {
+        const indicator = this._projectedIndicators[i];
+        if (indicator) {
+          this.setIndicatorAttributes(indicator, slide, i);
+          this.setIndicatorVisibility(indicator, slide.active);
+        }
+      });
+    }
```

Comment:
```suggestion
    this.requestUpdate()
```
See **indicatorTemplate**